byterun/dynlink.c

```diff
@@ -73,36 +73,43 @@ static c_primitive lookup_primitive(char * name)
 /* Parse the OCAML_STDLIB_DIR/ld.conf file and add the directories
    listed there to the search path */
 
-#define LD_CONF_NAME "ld.conf"
+#define LD_CONF_NAME _T("ld.conf")
 
-static char * parse_ld_conf(void)
+static charnat * parse_ld_conf(void)
 {
-  char * stdlib, * ldconfname, * config, * p, * q;
+  charnat * stdlib, * ldconfname, * wconfig, * p, * q;
+  char * config;
+#ifdef _WIN32
+  struct _stati64 st;
+#else
   struct stat st;
+#endif
   int ldconf, nread;
 
-  stdlib = caml_secure_getenv("OCAMLLIB");
-  if (stdlib == NULL) stdlib = caml_secure_getenv("CAMLLIB");
-  if (stdlib == NULL) stdlib = OCAML_STDLIB_DIR;
-  ldconfname = caml_stat_strconcat(3, stdlib, "/", LD_CONF_NAME);
-  if (stat(ldconfname, &st) == -1) {
+  stdlib = caml_secure_getenv(_T("OCAMLLIB"));
+  if (stdlib == NULL) stdlib = caml_secure_getenv(_T("CAMLLIB"));
+  if (stdlib == NULL) stdlib = caml_stat_strdup_to_utf16(OCAML_STDLIB_DIR); /* FIXME: free after use */
```

Comment:
This leak needs dealing with - a flag I guess?

Reply:
Yes, fixed.